docs/features/open-generic-names-in-nameof.md

```diff
@@ -0,0 +1,100 @@
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;~~,~~  
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;~~commas   ,~~
+
+
+unbound-type:  
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;type
+
+type-argument:  
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;type  
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&lt;empty&gt;
+
+
+The net effect of these grammar changes is that anywhere where we used to allow a generic-name, we now allow 'ommitted-type-arguments' to be present in the 'type-argument-list'.  i.e. *grammatically* the following are allowed:
+
+```c#
+List<> a;                       // should be error
+Dictionary<,> d;                // should be error
+Dictionary<int,> d;             // should be error
+new Dictionary<,>();            // should be error
+Method<>();                     // should be error
+Method<,>();                    // should be error
+Method<int,>();                 // should be error
+typeof(List<>);
+typeof(List<>.Enumerator);
+typeof(List<>.Add);
+nameof(List<>);
+nameof(List<>.Enumerator);
+nameof(List<>.Add);
+typeof(List<>[]);
+typeof(List<List<>>);           // should be error
+nameof(List<List<>>);           // should be error
+nameof(List<int>.Inner<>);      // should be error
+nameof(List<>.Inner<int>);      // should be error
+```
+
+Note: This is nearly identical to how the C# compiler *today* parses things.  It already uses unbound-type-arguments for error recovery purposes.   As such, these grammar changes need almost no changes to the C# parser.
+
+#### Semantic Changes  
+
+Of course, many of these we do not want to actually be allowed by the language as a whole.  This is where the "unbound-type" grammar-production comes in.  On the static-semantic side of things we  effectively want to say is that in all places in the langauge where we see "type" explicitly used, it is not ok to have an "omitted-type-argument".  The only times it is ok to have an ommitted type argument, is when the production starts with "unbound-type" or within a 'real' "nameof" expression.
+
+This will prevent these use cases:
+```c#
+List<> a;
+Dictionary<,> d;
+Dictionary<int,> d;
+new Dictionary<,>();
+Method<>();
+Method<,>();
+Method<int,>();
+```
+
+We still need to prevent cases like ```typeof(List<List<>>)``` though.  To enforce this we will require that type-argument-lists cannot contain omitted-type-arguments if they are contained within a type-argument-list with non-ommitted-type-argument.  Similarly, a type name cannot have a mix of type-arguments-lists with/without in a fully qualified name.  (Note: the compiler already does this today so it can error in this case for 'typeof').
+
+
+##### Semantic Model Details  
+This change produces some interesting cases for the semantic model.  For example (as raised by Neal) what should the SemanticModel produce for the following:
+
+```C#
+class KeyValuePair<A,B> 
+{
+}
+class G<T,U>
+{
+    public KeyValuePair<T,U> AProperty { get; }
+}
+class D<T> : G<T, int>

```

Comment:
:exclamation: The `T` defined here should be given a new name to eliminate ambiguity in referencing `G<T, int>` in the text below.